src/jdk.compiler/share/classes/com/sun/tools/javac/code/Lint.java

```diff
@@ -107,23 +133,48 @@ public Lint suppress(LintCategory... lc) {
         return l;
     }
 
+    // Compiler context
     private final Context context;
+    private final Options options;
 
     // These are initialized lazily to avoid dependency loops
     private Symtab syms;
     private Names names;
 
-    // Invariant: it's never the case that a category is in both "values" and "suppressedValues"
-    private final EnumSet<LintCategory> values;
-    private final EnumSet<LintCategory> suppressedValues;
+    // For the root instance only, these are initialized lazily
+    private EnumSet<LintCategory> values;           // categories enabled by default or "-Xlint:key" and not (yet) suppressed
+    private EnumSet<LintCategory> suppressedValues; // categories suppressed by augment() or suppress() (but not "-Xlint:-key")
```

Comment:
I was not expecting to find so many changes in this class. From the description I was expecting most changes in Options and some `small` changes in its clients

Reply:
Yep - those changes are mainly me trying to clean things up and improve the documentation and not directly related to this ticket.

I have another PR brewing that would be a better place for them so I'll revert those changes from this one - see 4e751f87d33.

Thanks for the feedback!